spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -219,87 +223,116 @@ public long executeDmlStatement(Statement statement) {
 
 	@Override
 	public void insert(Object object) {
-		applyMutations(this.mutationFactory.insert(object));
+		List<Mutation> mutations = this.mutationFactory.insert(object);
+		emitSaveEventsApplyMutations(mutations, Collections.singletonList(object), null);
```

Comment:
What's the purpose of the local variable `mutations`? Can't you just do it inline?